xlsynth/src/sv_bridge_builder.rs

```diff
@@ -55,12 +55,12 @@ fn is_screaming_snake_case(name: &str) -> bool {
     })
 }
 
-fn make_array_span_suffix(array_size: usize) -> String {
-    if array_size <= 1 {
-        "".to_string()
-    } else {
-        format!(" [{}:0]", array_size - 1)
+fn make_array_span_suffix(array_sizes: Vec<usize>) -> String {
+    let mut suffix = String::new();
+    for array_size in array_sizes.iter() {
+        suffix.push_str(&format!(" [{}:0]", array_size - 1));
```

Comment:
Instead of pushing onto a string can you push into a vec and at the end do `pieces.join(" ")`

Reply:
Also do you want to assert the array_sizes is non-empty for a precondition.

Reply:
did join("") because there's a missing space between 'logic' and arrays otherwise. and it looks hacky if I add extra " " at the beginning.

No assert since array_sizes is allowed to be empty, in which case it'll return an empty string